addons/point_of_sale/models/pos_config.py

```diff
@@ -321,6 +331,15 @@ def _onchange_start_category(self):
         if not self.start_category:
             self.iface_start_categ_id = False
 
+    @api.onchange('limit_categories', 'iface_available_categ_ids', 'iface_start_categ_id')
+    def _onchange_limit_categories(self):
+        if self.limit_categories:
+            if self.iface_start_categ_id not in self.iface_available_categ_ids:
+                self.iface_start_categ_id = False
+        else:
+            self.iface_available_categ_ids = False
+        self._compute_iface_start_categ_domain_ids()
+
```

Comment:
I think this is equivalent and a bit easier to read because it avoids the nested if:

```python
    @api.onchange('limit_categories', 'iface_available_categ_ids', 'iface_start_categ_id')
    def _onchange_limit_categories(self):
        if not self.limit_categories:
            self.iface_available_categ_ids = False
        elif self.iface_start_categ_id not in self.iface_available_categ_ids:
            self.iface_start_categ_id = False
```

Don't manually call `_compute_iface_start_categ_domain_ids`. With the correct dependencies like I suggested above the ORM recomputes automatically when `iface_available_categ_ids` changes.